Microsoft.Toolkit.Uwp.UI.Controls/Graph/AADLogin/AADLogin.cs

```diff
@@ -0,0 +1,210 @@
+
+        private Button _mainButton = null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AADLogin"/> class.
+        /// </summary>
+        public AADLogin()
+        {
+            DefaultStyleKey = typeof(AADLogin);
+            IsEnabled = false;
+        }
+
+        /// <summary>
+        /// Override default OnApplyTemplate to capture child controls
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            ApplyTemplate();
+
+            _mainButton = GetTemplateChild("btnMain") as Button;
+
+            AutomationProperties.SetName(_mainButton, SignInDefaultText);
+
+            _mainButton.Click += async (object sender, RoutedEventArgs e) =>
+            {
+                var btn = sender as Button;
+
+                if (string.IsNullOrEmpty(CurrentUserID))
+                {
+                    btn.IsEnabled = false;
+                    if (await SignInAsync())
+                    {
+                        AutomationProperties.SetName(_mainButton, string.Empty);
+                        btn.Flyout = GenerateMenuItems();
+                    }
+
+                    btn.IsEnabled = true;
+                }
+            };
+        }
+
+        /// <summary>
+        /// This method is used to prompt to login screen.
+        /// </summary>
+        /// <returns>True if sign in successfully, otherwise false</returns>
+        public async Task<bool> SignInAsync()
+        {
+            if (!string.IsNullOrEmpty(ClientId) && !string.IsNullOrEmpty(Scopes))
+            {
+                var token = await GetTokenForUserAsync();
+
+                if (!string.IsNullOrEmpty(token))
+                {
+                    var graphClient = Common.GetAuthenticatedClient(token);
+
+                    GraphAccessToken = token;
+
+                    CurrentUserID = (await graphClient.Me.Request().GetAsync()).Id;
+
+                    OnSignInCompleted(new SignInEventArgs()
+                    {
+                        GraphClient = graphClient,
+                        GraphAccessToken = token,
+                        CurrentSignInUserId = CurrentUserID
+                    });
+
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// This method is used to sign out the currently signed on user
+        /// </summary>
+        public void SignOut()
+        {
+            if (_identityClientApp.Users != null)
+            {
+                foreach (var user in _identityClientApp.Users)
+                {
+                    _identityClientApp.Remove(user);
+                }
+
+                CurrentUserID = string.Empty;
+                _mainButton.Flyout = null;
+                OnSignOutCompleted();
+            }
+
+            AutomationProperties.SetName(_mainButton, SignInDefaultText);
+        }
+
+        private void InitialPublicClientApplication()
```

Comment:
Rename to `InitializePublicClientApplication`

Reply:
Done and committed.